internal/report/github/summary.go

```diff
@@ -0,0 +1,127 @@
+package github
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/report"
+)
+
+type Reporter struct {
+	beginTime       time.Time
+	stepSummaryFile string
+	results         []report.TestResult
+}
+
+func NewJobSummaryReporter() Reporter {
+	fmt.Printf("GITHUB_STEP_SUMMARY: %s", os.Getenv("GITHUB_STEP_SUMMARY"))
+	return Reporter{
+		beginTime:       time.Now(),
+		stepSummaryFile: os.Getenv("GITHUB_STEP_SUMMARY"),
+	}
+}
+
+func (r *Reporter) isActive() bool {
+	return r.stepSummaryFile != ""
+}
+
+func (r *Reporter) Add(t report.TestResult) {
+	if !r.isActive() {
+		return
+	}
+	r.results = append(r.results, t)
+}
+
+func hasSomeDevice(results []report.TestResult) bool {
+	for _, t := range results {
+		if t.DeviceName != "" {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *Reporter) Render() {
+	if !r.isActive() {
+		return
+	}
+
+	endTime := time.Now()
+	hasDevices := hasSomeDevice(r.results)
+	errors := 0
+	inProgress := 0
+
+	content := renderHeader(hasDevices)
+	for _, result := range r.results {
+		if result.Status == "in progress" {
+			inProgress++
+		}
+		if result.Status == "failed" {
```

Comment:
1. Does that cover all the states? See [table reporter](https://github.com/saucelabs/saucectl/blob/5f462ab1c3383283576a15fd16ed9161dceea078/internal/report/table/table.go#L106-L106).
2. Would be safer to use a reference to the states declared in the packages _job_ and _imagerunner_ than using an adhoc string. Easier to refactor later if we need to.